### PR TITLE
[dev] Added env variable to append other custom command location

### PR DESCRIPTION
**Pull request by franramirez688**
Changelog: omit
Docs: omit

### CODE REVIEW

File: conan/cli/cli.py [hunk 1]
```diff
@@ -47,35 +47,41 @@ def _add_commands(self):
             for k, v in self._commands.items():  # Fill groups data too
                 self._groups[v.group].append(k)
 
-        custom_commands_path = HomePaths(self._conan_api.cache_folder).custom_commands_path
-        if not os.path.isdir(custom_commands_path):
-            return
+        conan_custom_commands_path = HomePaths(self._conan_api.cache_folder).custom_commands_path
+        # Important! This variable should be only used for testing/debugging purpose
+        developer_custom_commands_path = os.getenv("_CONAN_INTERNAL_CUSTOM_COMMANDS_PATH")
+        custom_commands_folders = [developer_custom_commands_path, conan_custom_commands_path] \
```
Comment (AbrilRBS) [inline conan/cli/cli.py:55]: This and the test below confuse me a bit, let's check

**Review by AbrilRBS — COMMENTED** [review]
🤔

File: conan/cli/cli.py (hunk 1 above)
Comment (memsharded) [inline conan/cli/cli.py:55]: I think the order should be reversed, to give priority to the developer ones

**Review by franramirez688 — COMMENTED** [review]
@memsharded @RubenRBS @czoido 

The order is confusing but it's fine as it is, and the test is already checking that. I just added a comment to explain better what's happening under the hood. Basically, we're using `sys.path.append`, so the developer ones will have preference over any other custom commands defined.

**Review by memsharded — APPROVED** [review]

**Review by czoido — APPROVED** [review]

**Pull request merged**